Automattic/jetpack
# Share Post: initial approach

This PR is the first approach of the Share Post button of the Publicize feature.

Related to https://github.com/Automattic/jetpack/issues/21087

## Feature flag

The RePublicize feature is disabled by default. In order to enable it, use the filter:

```php
add_filter( 'jetpack_block_editor_republicize_feature', '__return_true' );
```

## UI

The UI suggested here is a design in between the current Publicize and a possible new RePublicize/ReScheduling feature.

### Significant changes

#### `Share is enable` toggle control

It allows enabling/disabling the sharing/re-sharing feature. It worths mentioning the state is stored in the post metadata.

<img width="282" alt="Screen Shot 2021-10-06 at 10 58 50 AM" src="https://user-images.githubusercontent.com/77539/136217570-c53bc581-7189-4e11-99c8-8a641047831d.png">

#### Publicize description based on sharing and the post state.

Unpublished / Unshared / Disabled | Unpublished / Unshared / Enabled  |  Published /  Disabled | Published /  Enabled
-------|-------|----|---
![image](https://user-images.githubusercontent.com/77539/136217650-f18d57e6-9f49-449c-9b39-97be2ed522b2.png) | ![image](https://user-images.githubusercontent.com/77539/136217910-f1398b10-174f-4a0c-9b87-29edeb1cb07c.png) | ![image](https://user-images.githubusercontent.com/77539/136218633-8045d65e-a6e8-4635-9852-91874c2926ed.png) | ![image](https://user-images.githubusercontent.com/77539/136218491-85974983-fb16-426b-9d7e-64a3a89a572c.png)

#### Share post button

This button will be visible only when the post has been published

Feature/connections disabled | Feature/connections enabled | Sharing
-----|------|------
![image](https://user-images.githubusercontent.com/77539/136218965-5da5f7d7-ed5b-45a4-be31-ac78403da30b.png) | ![image](https://user-images.githubusercontent.com/77539/136219142-3b475eaa-790c-4247-b069-697323f8887f.png) | <img width="281" alt="Screen Shot 2021-10-06 at 11 08 38 AM" src="https://user-images.githubusercontent.com/77539/136219444-6dd42a56-0b95-40b5-b29e-a3d301a47bfd.png">

#### Notices

When the post is properly shared, the app shows a notice (snake mode) providing a link to see the shared post.

<img src="https://user-images.githubusercontent.com/77539/136219996-90305c62-3e3b-445f-b9a6-f49af6a61c6c.png" width="300px" />

When something worng happens, the app shows an error notice:

<img src="https://user-images.githubusercontent.com/77539/136220333-a07a5cba-48ec-4992-aabf-9be3f4fa249d.png" width="600px" />


### 🍿 Video

https://user-images.githubusercontent.com/77539/136221352-288030d4-7a5d-4e8b-b83b-0816a967c1f9.mov


#### Changes proposed in this Pull Request:

* Publicize: Add share post button

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Apply these changes
* Enable the feature:
```php
add_filter( 'jetpack_block_editor_republicize_feature', '__return_true' );
```
* Check it's possible to re-share the post by clicking on the share post button.
